src/device.rs

```diff
@@ -4,11 +4,16 @@
 
 use gdk_sys;
 use glib::translate::*;
+use glib::object::IsA;
 use AxisUse;
 use Device;
 
-impl Device {
-    pub fn get_axis(&self, axes: &mut [f64], use_: AxisUse, value: &mut f64) -> bool {
-        unsafe { from_glib(gdk_sys::gdk_device_get_axis(self.to_glib_none().0, axes.as_mut_ptr(), use_.to_glib(), value)) }
+pub trait DeviceExtManual: 'static {
+    fn get_axis(&self, axes: &mut [f64], use_: AxisUse, value: &mut f64) -> bool;
+}
+
+impl<O: IsA<Device>> DeviceExtManual for O {
```

Comment:
`gdk::Device` is not subclassable outside of `gdk` and there are no subclasses inside `gdk`, are there?